HACKING.rst

```diff
@@ -1,263 +1,512 @@
-- The docs use ``reStructuredText`` for markup. See a live demo at
-  http://rst.ninjs.org/
-- The help information on each module or state is culled from the python code
-  that runs for that piece. Find them in ``salt/modules/`` or ``salt/states/``.
-- If you are developing using Arch Linux (or any other distribution for which
-  Python 3 is the default Python installation), then ``sphinx-build`` may be
-  named ``sphinx-build2`` instead. If this is the case, then you will need to
-  run the following ``make`` command::
+::
 
-    make SPHINXBUILD=sphinx-build2 html
+   brew update
+   brew install pyenv
+   brew install pyenv-virtualenv
 
-Installing Salt for development
--------------------------------
+--------------
 
-Clone the repository using::
+Now add pyenv to your ``.bashrc``:
 
-    git clone https://github.com/saltstack/salt
-    cd salt
+::
 
-.. note:: tags
+   echo 'export PATH="$HOME/.pyenv/bin:$PATH"' >> ~/.bashrc
+   pyenv init 2>> ~/.bashrc
+   pyenv virtualenv-init 2>> ~/.bashrc
 
-    Just cloning the repository is enough to work with Salt and make
-    contributions. However, you must fetch additional tags into your clone to
-    have Salt report the correct version for itself. To do this, fetch the tags
-    with the command::
+For other shells, see `the pyenv
+instructions <https://github.com/pyenv/pyenv#basic-github-checkout>`__.
 
-        git fetch --tags
+Go ahead and restart your shell. Now you should be able to install a new
+version of Python:
 
-Preparing your system
-~~~~~~~~~~~~~~~~~~~~~
+::
+
+   pyenv install 3.7.0
+
+If that fails, don’t panic! You’re probably just missing some build
+dependencies. Check out `pyenv common build
+problems <https://github.com/pyenv/pyenv/wiki/Common-build-problems>`__.
+
+Now that you’ve got your version of Python installed, you can create a
+new virtual environment with this command:
+
+::
+
+   pyenv virtualenv 3.7.0 salt
+
+Then activate it:
+
+::
+
+   pyenv activate salt
+
+Sweet! Now you’re ready to clone salt so you can start hacking away!
+
+Cloning Salt
+~~~~~~~~~~~~
+
+Clones are so shallow. Well, this one is anyway:
+
+::
+
+   git clone --depth=1 --origin salt https://github.com/saltstack/salt.git
+
+This creates a shallow clone of salt, which should be fast. Most of the
+time that’s all you’ll need, and you can start building out other
+commits as you go. If you *really* want all 108,300+ commits you can
+just run ``git fetch --unshallow``. Then go make a sandwich because it’s
+gonna be a while.
+
+Fork Salt
+~~~~~~~~~
+
+You’re also going to want to head over to GitHub and create your own
+`fork of salt <https://github.com/saltstack/salt/fork>`__. Once you’ve
+got that setup you can add it as a remote:
+
+::
+
+   git remote add yourname <YOUR SALT REMOTE>
+
+If you use your name to refer to your fork, and ``salt`` to refer to the
+official Salt repo you’ll never get ``upstream`` or ``origin`` confused.
+
+Install sphinx
```

Comment:
Same as the other doc, and, why are we repeating doc content?

Reply:
I can remove the markdown doc -- it's easier to write than .rst and I can just use pandoc to convert :trollface:

Reply:
Maybe, instead of duplicating content, `HACKING.md` should be minimal and direct to online documentation?
Ie, a single source of truth to maintain?

Reply:
I was under the impression that HACKING.rst existed because our existing online documentation was reading it as a .rst, but it appears that https://docs.saltstack.com/en/latest/topics/development/hacking.html is different than HACKING, so... ew, no. Apparently I'll be removing HACKING.rst and moving the content, since we're not including it anywhere.

To be fair, I wasn't *duplicating* content, I was (com|trans)piling it to rst, exactly the same as how we compile our manpages.

I do prefer a single source of truth, but I also like enabling contributions for people who aren't interested in being constantly online.

We *do* build our online docs from the repo, though, so it's possible to point to a location within the repo as well :thinking: 

Yeah, I'll go ahead and do that maybe later today, or tomorrow if I can't get to it.

Reply:
:+1:

Reply:
We probably need to define what the scope of each community doc contains, where it lives, and then where there is duplicate content just link to that content in the other doc. I'm meeting with stakeholders next week to talk about Community Docs strategy and think both of you would be great to have as members of an advisory board to help answer these questions.